micrometer-core/src/main/java/io/micrometer/core/instrument/binder/UptimeMetrics.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder;
+
+import static java.util.Objects.requireNonNull;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+
+import io.micrometer.core.instrument.MeterRegistry;
+
+/**
+ * Uptime metrics.
+ *
+ * @author Michael Weirauch
+ */
+public class UptimeMetrics implements MeterBinder {
+
+    private final RuntimeMXBean runtimeMXBean;
+
+    public UptimeMetrics() {
+        this(ManagementFactory.getRuntimeMXBean());
+    }
+
+    // VisibleForTesting
+    UptimeMetrics(RuntimeMXBean runtimeMXBean) {
+        this.runtimeMXBean = requireNonNull(runtimeMXBean);
+    }
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+        registry.gauge("uptime", runtimeMXBean, (x) -> Long.valueOf(x.getUptime()).doubleValue());
```

Comment:
I'm not opposed to this metric, but do want to point out the Prometheus approach is to report the process start time (see https://github.com/prometheus/client_java/blob/master/simpleclient_hotspot/src/main/java/io/prometheus/client/hotspot/StandardExports.java#L72)

The logic being that the process start time won't change and the query can calculate the uptime based on that. This is also the typical approach for all timestamps with Prometheus, since you want the metric to report when something happened and the query can determine the 'time since' logic.

Reply:
I was tempted to add "starttime" additionally in the beginning. The thing is only "starttime" is crucial to those math-agnostic metric-stores like CloudWatch.
Is adding both (uptime,starttime) in two MeterBinders an option, so people can choose what they wan't/need?

Regarding resolution/units: Should one drop precision just because Promtheus' time() returns unix-time (seconds) instead of millis? (micrometer != prometheus)

Reply:
> Should one drop precision just because Promtheus' time() ...

The gauge value will be reported as a floating-point value, so no precision will be lost. I'm working on support for scaling time-based gauges now for this PR.

Reply:
Maybe I shouldn't say _no_ precision is lost, because at a sufficient scale, the cumulative error of floating point math will add up. Apparently this is something Prometheus devs are willing to accept, and I'm happy to accommodate for as long as it doesn't cause that community any heartburn.

Reply:
> Is adding both (uptime,starttime) in two MeterBinders an option

My vote would be to just add them both to the same binder. It's only one additional metric which can be ignored if not needed. Your statement that starttime is crucial to CloudWatch has me a bit alarmed: I _really_ need to look at CloudWatch before the 1.0 release (#66).